container/src/main/java/com/flipkart/poseidon/core/PoseidonServlet.java

```diff
@@ -121,7 +122,7 @@ protected void doRequest(HttpMethod method, HttpServletRequest httpRequest, Http
             StringBuffer requestBuffer = new StringBuffer();
             String line = null;
             try {
-                BufferedReader reader = httpRequest.getReader();
+                BufferedReader reader = new BufferedReader(new InputStreamReader(httpRequest.getInputStream()));
```

Comment:
I thought this change alone would be sufficient. Let me understand the need of this along with RequestReplayFilter. Lets sync up offline.